bin/i18n-update.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+# 
+# Handles the process of updating the i18n localizations of plugins, including Gutenberg.
+# The main goals of this command are:
+#
+#   1. Download translation files and optimize them by filtering out the unused strings,
+#      previously extracted from the source map files of React Native bundle.
+#
+#      This step produces the following output files:
+#      - src/i18n-cache/{PLUGIN_NAME}/data/{LOCALE}.json     [Translation files]
+#      - src/i18n-cache/{PLUGIN_NAME}/data/index.js          [JS file to import translations]
+#
+#   2. Generate localization strings files that include the strings only used in native JS
+#      source code files ("native.js", "ios.js or "android.js"). The translations of these
+#      strings are not included in the GlotPress projects of their plugins hence, they require
+#      to be requested as part of the main apps’ strings.
+#
+#      This step produces the following output files:
+#      - bundle/android/strings.xml                     [Localization strings files for Android platform]
+#      - bundle/ios/GutenbergNativeTranslations.swift   [Localization strings files for iOS platform]
+
+# Exit if any command fails
+set -e
+
+# Get arguments
+while test $# -gt 0; do
+  case "$1" in
+    -h|--help)
+      echo "options:"
+      echo "-h, --help    show brief help"
+      echo "-p, --path    local path for generating files (by default a temp folder will be used)"
+      echo "-d, --debug   print extra info for debugging"
+      exit 0
+      ;;
+    -p|--path*)
+      shift
+      LOCAL_PATH=$1
+      shift
+      ;;
+    -d|--debug*)
+      shift
+      DEBUG='true'
+      ;;
+    *)
+      break
+      ;;
+  esac
+done
+
+function error() {
+  echo -e "\033[0;31m$1\033[0m"
+  exit 1
+}
+
+function fetch_translations() {
+  local PLUGIN_NAME=$1
+  local OUTPUT_PATH=$2
+  local USED_STRINGS_FILE=$3
+
+  echo -e "\n\033[1mDownload I18n translations for \"$PLUGIN_NAME\" plugin\033[0m"
+  node gutenberg/packages/react-native-editor/bin/i18n-translations-download $PLUGIN_NAME $OUTPUT_PATH $USED_STRINGS_FILE
+
+  if [[ "$PLUGIN_NAME" == "gutenberg" ]]; then
+    echo "Update \"react-native-editor\" package i18n cache"
+    cp -r $OUTPUT_PATH/gutenberg/data gutenberg/packages/react-native-editor/i18n-cache
+    cp $OUTPUT_PATH/gutenberg/index.js gutenberg/packages/react-native-editor/i18n-cache/index.native.js
+  fi
+}
+
+# Set target path
+if [[ -n $LOCAL_PATH ]]; then
+  TARGET_PATH=$LOCAL_PATH
+else
+  TARGET_PATH=$(mktemp -d)
```

Comment:
Probably should add a trap to clean up the temp file :  
something like `trap 'rm -f "$TARGET_PATH"' EXIT`. 
That will run when the script exits ( well except on `kill -9` )

Reply:
Oh I didn't know about that command 🤩 , I added this in https://github.com/wordpress-mobile/gutenberg-mobile/pull/4366/commits/a9cc79e9051e094d54dd161d2d576c9513311cb3.

> That will run when the script exits ( well except on kill -9 )
I don't think this should be a major issue as it's a temporary folder and eventually it will be deleted by the OS.